platform/ios/MGLAccountManager.m

```diff
@@ -0,0 +1,50 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLAccountManager.h"
+#import "NSProcessInfo+MGLAdditions.h"
+#import "MGLMapboxEvents.h"
+
+@interface MGLAccountManager()
+
+@property (atomic) NSString *accessToken;
+
+@end
+
+
+@implementation MGLAccountManager
+
+static MGLAccountManager *_sharedManager;
+
+// Can be called from any thread. Called implicitly from any
+// public class convenience methods.
+//
++ (instancetype) sharedInstanceWithAccessToken:(NSString *)token {
```

Comment:
> `[[MGLAcountManager sharedManager] setAccessToken:@"FOO"]`

The way we do this in MBXMapKit is more brief: 

> `[MGLAccountManager setAccessToken:]`

Which is basically just a shortcut to the above, but doesn't require the user to worry about shared instances.